app/src/main/java/org/simple/clinic/user/UserSession.kt

```diff
@@ -114,6 +128,7 @@ class UserSession @Inject constructor(
 
   fun findExistingUser(phoneNumber: String): Single<FindUserResult> {
     return registrationApi.findUser(phoneNumber)
+        .flatMap { saveUserAsNotSignedIn(it).toSingleDefault(it) }
```

Comment:
This looks like a hidden behavior. Can you either change the function name or move this to somewhere else?

Reply:
👍